addons/account_payment/static/src/js/payment_form.js

```diff
@@ -15,12 +15,12 @@ odoo.define('account_payment.payment_form', require => {
          *
          * @override method from payment.payment_form_mixin
          * @private
-         * @param {string} provider - The provider of the selected payment option's acquirer.
+         * @param {string} code - The provider code of the selected payment option.
          * @param {number} paymentOptionId - The id of the selected payment option.
          * @param {string} flow - The online payment flow of the selected payment option.
          * @return {object} The extended transaction route params.
          */
-        _prepareTransactionRouteParams: function (provider, paymentOptionId, flow) {
+        _prepareTransactionRouteParams: function (code, paymentOptionId, flow) {
```

Comment:
This is an override, changing provider => code here would imply changing it in all the other JS files too, this isn't the only method nor the only files that have it. I only commented once here for these because I'm not sure that we want to change it in all the JS too, but that'd makes sens. But if you change here, change all the other JS files too.

Reply:
you're right, it's done everywhere now :)